src/ModalForm.js

```diff
@@ -0,0 +1,109 @@
+import React, {PropTypes} from 'react';
+import {
+  Button,
+  Modal,
+  FormGroup,
+  ControlLabel,
+  FormControl,
+  HelpBlock
+} from 'react-bootstrap';
+
+function FieldGroup({ id, label, help, ...props }) {
+  return (
+    <FormGroup controlId={id}>
+      <ControlLabel>{label}</ControlLabel>
+      <FormControl {...props} />
+      {help && <HelpBlock>{help}</HelpBlock>}
+    </FormGroup>
+  );
+}
+
+const ModalForm = React.createClass({
+
+  getInitialState() {
+    return {
+      showModal: false,
+    };
+  },
+
+  close() {
+    this.setState({ showModal: false });
+  },
+
+  open() {
+    this.setState({ showModal: true });
+  },
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const {title, description, buttonText, url} = this.state;
```

Comment:
I think there are two ways to get data out, one is to track them internally in the Modal Form's state and then pass them to the onSubmit function like you're doing, or to pass them down as a prop from the app's state and then pass them in.